src/index.js

```diff
@@ -79,6 +78,25 @@ async function start(fields, doRetry = true) {
       googleUtils
     )
 
+    // update the contact account
+    const lastLocalSync = new Date().toISOString()
+    let nextSyncToken = syncTokenBefore
+    // if we changed something on google, we have to ask for a new sync token
+    if (Object.values(result.google).some(v => v !== 0)) {
+      const response = await googleUtils.getAllContacts({
+        syncToken: syncTokenBefore
+      })
+      nextSyncToken = response.nextSyncToken
```

Comment:
My worry is that if we'r reaching this code, there may have been other changes on google side in between `syncTokenBefore ` and `nextSyncToken `, which we would discard.

I don't know how probable this is, I guess it depends on the execution time. I suspect it would be ok, but we should at least add a comment about it.

Ideally, I guess we should check that the contacts we get here are the same ones that we updated in the main loop.

Reply:
yeah, I thought about it too. I will add a comment. We could check the sum of created and updated contacts and check that it is equal to the length of the contacts in the response

Reply:
Ah yes, we could compare the sums, and if they differ we don't change the token ? We might lose some cozy changes, but we'll save the google changes, which is generally what we prefer !

Reply:
Discussed with @Crash-- 
TODO:
- [ ] check the count vs length
- [ ] log if they are different
- [ ] inform Thomas about this

Reply:
Works for me!